src/typeahead-standalone.ts

```diff
@@ -448,6 +450,12 @@ export default function typeahead<T extends Dictionary>(config: typeaheadConfig<
     if (prefetch && prefetch.when === 'onFocus') {
       prefetchData();
     }
+    if (fnSource !== null) {
+      // if the source is a function then run on focus
+      trie.clear()
```

Comment:
As your clear the trie, it means that it would be impossible to use `fnSource` in combination with any of the other sources(local/prefetch/remote). For example, what if someone wanted to prefetch data ? In the above code, even though the data would be prefetched, we wouldn't see it since the trie was cleared. 

It it absolutely necessary to clear the trie ? Because the `fnSource` would also work without clearing the trie (the order of the results might change though)

Reply:
Good point about how the clear makes it impossible to use other sources.  Honestly, I only see a use case for "pure local, or pure function or pure remote or remote with prefetch". But that is my bias.  Nonetheless, I do not wish to impose this on others and I don't want to make a breaking change.  Yet, with the use of the function source on the focus event means the second and third times the input gets focus the previous results from the function source are still in the `trie`.

Reply:
Hmmm. I get your point. Well, in that case, I think `fnSource` should be renamed to `customSource` and it should be specified in the readme/documentation that while using this customSource, it would not be possible to use other sources in conjunction with this one and that results returned by the customSource would not be cached automatically by the library thereby requiring the end user to generate results each time depending upon their application context. In doing this, we will ensure that it will not be a breaking change either.